src/Components/LifecycleChart/lifecycleChart.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import '@patternfly/react-core/dist/styles/base.css';
+import { Chart, ChartAxis, ChartBar, ChartGroup, ChartLabel, ChartTooltip, ChartVoronoiContainer } from '@patternfly/react-charts';
+import {Card} from '@patternfly/react-core'; 
+
+
+const LifecycleChart: React.FC<React.PropsWithChildren> = () => {
+
+    // Start = y0, end = y
+    const alerts = [
+      [
+        {x: "Node.js 16", y0: new Date("2023-01"), y: new Date("2024-06"), packageType: 'Retired' }
+      ],
+      [
+        {x: "gcc-toolset 12", y0: new Date("2023-01"), y: new Date("2025-10"), packageType: 'Support ends within 6 months' }
+      ],
+      [
+        {x: "Ruby 3.1", y0: new Date("2024-08"), y: new Date("2025-06"), packageType: 'Not installed' }
+      ],
+      [
+        {x: "gcc-toolset 12", y0: new Date("2023-01"), y: new Date("2027-10"), packageType: 'Supported' }
+      ],
+      [
+        {x: "Ruby 3.0", y0: new Date("2024-08"), y: new Date("2025-06"), packageType: 'Upcoming release' }
+      ],      
+    ];
+
+
+
+    const formatDate = (date: Date, isTime: boolean) => {
+      const dateString = date?.toLocaleDateString("en-US", { year: 'numeric' });
+      const timeString = date?.toLocaleTimeString("en-US", { hour12: false });
+      return isTime ? `${dateString} ${timeString}` : dateString;
+    };
+
+    const getChart = (alert: any[], index: number) => {
+      const data: any[] = [];
+
+      alert?.map((datum: { packageType: string; }) => {
+        data.push({
+          ...datum,
+          x: alerts.length - index,
+          fill: datum.packageType === "Retired"
```

Comment:
What I would suggest here is a call to a separate function with a switch statement based on datum.packageType, like this: 

```
const getPackageColor = () => {
     switch(datum.packageType) {
        case 'Retired':
          return "var(--pf-v5-global--danger-color--100)";
        // ... other cases go here
        case default:
          return;
     } 
}
```

The ternary statements work, but this would be cleaner and easier to modify if the list of colors changes/grows.

Reply:
Sounds good i'll make this change